systemtest/src/test/java/io/strimzi/systemtest/upgrade/OlmUpgradeST.java

```diff
@@ -0,0 +1,182 @@
+            OlmResource.clusterOperator(namespace, OlmInstallationStrategy.Manual, "v" + getFirstSupportedItemFromUpgradeJson().getString("fromVersion"));
+        }
+
+        // In chainUpgrade we want to setup Kafka only at the start and then upgrade it via CO
+        if (KafkaResource.kafkaClient().inNamespace(namespace).withName(CLUSTER_NAME).get() == null) {
+            JsonObject firstSupportedItemFromUpgradeJsonArray = getFirstSupportedItemFromUpgradeJson();
+            String url = firstSupportedItemFromUpgradeJsonArray.getString("urlFrom");
+            File dir = FileUtils.downloadAndUnzip(url);
+
+            // In chainUpgrade we want to setup Kafka only at the begging and then upgrade it via CO
+            kafkaYaml = new File(dir, firstSupportedItemFromUpgradeJsonArray.getString("fromExamples") + "/examples/kafka/kafka-persistent.yaml");
+            LOGGER.info("Going to deploy Kafka from: {}", kafkaYaml.getPath());
+            KubeClusterResource.cmdKubeClient().create(kafkaYaml);
+            // Wait for readiness
+            waitForReadinessOfKafkaCluster();
+
+            OlmResource.getClosedMapInstallPlan().put(OlmResource.getNonUsedInstallPlan(), Boolean.TRUE);
+
+            ResourceManager.setMethodResources();
+        }
+
+        kafkaBasicClientJob.producerStrimzi().done();
+        kafkaBasicClientJob.consumerStrimzi().done();
+
+        String clusterOperatorDeploymentName = ResourceManager.kubeClient().getDeploymentNameByPrefix(Environment.OLM_OPERATOR_DEPLOYMENT_NAME);
+        LOGGER.info("Old deployment name of cluster operator is {}", clusterOperatorDeploymentName);
+
+        // ======== Cluster Operator upgrade starts ========
+        Map<String, String> kafkaSnapshot = StatefulSetUtils.ssSnapshot(KafkaResources.kafkaStatefulSetName(CLUSTER_NAME));
+
+        // wait until non-used install plan is present (sometimes install-plan did not append immediately and we need to wait for at least 10m)
+        OlmUtils.waitUntilNonUsedInstallPlanIsPresent(toVersion);
+
+        // Cluster Operator
+        OlmResource.upgradeClusterOperator();
+
+        // wait until RU is finished (first run skipping)
+        StatefulSetUtils.waitTillSsHasRolled(KafkaResources.kafkaStatefulSetName(CLUSTER_NAME), 3, kafkaSnapshot);
+        // ======== Cluster Operator upgrade ends ========
+
+        clusterOperatorDeploymentName = ResourceManager.kubeClient().getDeploymentNameByPrefix(Environment.OLM_OPERATOR_DEPLOYMENT_NAME);
+        LOGGER.info("New deployment name of cluster operator is {}", clusterOperatorDeploymentName);
+        ResourceManager.setCoDeploymentName(clusterOperatorDeploymentName);
+
+        // verification that cluster operator has correct version (install-plan) - strimzi-cluster-operator.v[version]
+        String afterUpgradeVersionOfCo = OlmResource.getClusterOperatorVersion();
+
+        // if HEAD -> 6.6.6 version
+        toVersion = toVersion.equals("HEAD") ? "6.6.6" : toVersion;
```

Comment:
I think you should use Env var for OLM version instead of the hardcoded random version which we use in our automation.

Reply:
Done